Bundle/CriteriaBundle/DataSource/DomainNameDataSource.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Victoire\Bundle\CriteriaBundle\DataSource;
+
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class DomainNameDataSource.
+ */
+class DomainNameDataSource
+{
+    /**
+     * @var array
+     */
+    private $domainName;
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * DomainNameDataSource constructor.
+     *
+     * @param array        $domainName
+     * @param RequestStack $requestStack
+     */
+    public function __construct(
+        array $domainName,
+        RequestStack $requestStack
+    ) {
+        $this->domainName = $domainName;
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCurrentDomainName()
+    {
+        return $this->requestStack->getCurrentRequest()->getHost();
+    }
+
+    /**
+     * @return array
+     */
+    public function getDomainName()
+    {
+        return $this->domainName[0];
```

Comment:
Why do you take only the first item ? This should be an array of all the config defined domain names (if so, it would be better to use plural)